presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -343,4 +365,18 @@ private boolean isSchemaOwner(ConnectorSecurityContext context, String schemaNam
         }
         return false;
     }
+
+    private boolean checkSchemaAnyPermission(ConnectorSecurityContext context, String schemaName)
+    {
+        for (TableAccessControlRule rule : tableRules) {
+            Optional<Set<TablePrivilege>> tablePrivileges = rule.matchSchema(context.getIdentity().getUser(), schemaName);
+            if (tablePrivileges.isPresent()) {
+                if (tablePrivileges.get().isEmpty()) {
+                    return false;
```

Comment:
schema is visilble if user has any access to any table (not the first table in the list)

Reply:
should we just check the first matched rule ? this is to align with the logic check table permission.
/test/resource/table.json secret rule is the very first (allow no privilege) 
even the second admin rule match for any table/schema, admin should have no access to table in secret schema, hence no access to secret schema, correct ?

Reply:
> this is to align with the logic check table permission

But you don't check table permission, but schema. And even if user has no access to given table, they can still see the schema, because they could have access to other table.

Please make sure you have a test for such case.

Reply:
1. I checked table permission should be non-empty
2. user could provide table patterns in rule (table1|table2|table3)
3. yes, maybe user add additional table rule for other table pattern (table_a|table_b|table_c) for the same schema. However, this may contradict previous rule (example: 'secret' schema)
I think filter schema based on table rules is complex than we thought, we need to merge those rules somehow.
I am suggesting filter schema with schema rule, keep independent and simple @kokosing what do you think ?